File: services/identity-service/src/main/java/br/com/deltaglobalbank/identity/domain/tenant/TenantStatus.java
package br.com.deltaglobalbank.identity.domain.tenant;

import java.util.Arrays;

public enum TenantStatus {
    ACTIVE,
    SUSPENDED,
    INACTIVE;

    public String toDatabaseValue() {
        return name().toLowerCase();
    }

    public static TenantStatus fromDatabaseValue(String value) {
        return Arrays.stream(values())
            .filter(status -> status.name().equalsIgnoreCase(value))
            .findFirst()
            .orElseThrow(() -> new IllegalArgumentException("Status de tenant inválido: " + value));
    }
}

File: services/identity-service/src/main/kotlin/br/com/deltaglobalbank/identity/domain/tenant/TenantStatus.kt
package br.com.deltaglobalbank.identity.domain.tenant

enum class TenantStatus {
    ACTIVE,
    SUSPENDED,
    INACTIVE;

    fun toDatabaseValue(): String = name.lowercase()

    companion object {
        fun fromDatabaseValue(value: String): TenantStatus =
            entries.firstOrNull { it.name.equals(value, ignoreCase = true) }
                ?: throw IllegalArgumentException("Status de tenant inválido: $value")
    }
}